src/streaming/controllers/PlaybackController.js

```diff
@@ -104,21 +101,35 @@ function PlaybackController() {
         }
     }
 
-    function onPeriodSwitchStarted(e) {
-        if (!isDynamic && e.fromStreamInfo && earliestTime[e.fromStreamInfo.id] !== undefined) {
-            delete bufferedRange[e.fromStreamInfo.id];
-            delete earliestTime[e.fromStreamInfo.id];
+    function onStreamInitialized(e) {
+        // Seamless period switch
+        if (streamSwitch && isNaN(streamSeekTime)) return;
+
+        // Seek new stream in priority order:
+        // - at seek time (streamSeekTime) that required period switch
+        // - at start time provided in URI parameters
+        // - at stream/period start time (for static streams) or live start time (for dynamic streams)
+        let startTime = streamSeekTime;
+        if (isNaN(startTime)) {
+            startTime = isDynamic ? e.liveStartTime : e.streamInfo.start;
+            startTime = Math.max(startTime, getStartTimeFromUri());
+            if (startTime === videoModel.getTime()) return;
         }
+
+        // Trigger PLAYBACK_SEEKING event for controllers
+        eventBus.trigger(Events.PLAYBACK_SEEKING, {
+            seekTime: startTime
+        });
+        // Seek video model
+        seek(startTime, false, true);
```

Comment:
can startTime be NaN here? Avoid seeking with NaN

Reply:
should not, the only case it can be NaN is for live stream, if a request can't be found at liveEdge (see StreamProcessor::findRequestForLiveEdge())
But yes, we can add a check to avoid seeking with NaN